src/ui/screens/AddBountyScreen.tsx

```diff
@@ -0,0 +1,160 @@
+import React, {useEffect, useMemo, useRef} from 'react';
+import {StyleSheet, View} from 'react-native';
+import {NavigationProp} from '@react-navigation/native';
+import {Modalize} from 'react-native-modalize';
+import {AppStackParamList} from '@ui/navigation/navigation';
+import {Headline, TextInput, Button, Text, useTheme} from '@ui/library';
+import {Layout} from '@ui/components/Layout';
+import {standardPadding} from '@ui/styles';
+import {Account} from 'src/api/hooks/useAccount';
+import {SelectAccount} from '@ui/components/SelectAccount';
+import {Padder} from '@ui/components/Padder';
+import {useApiTx} from 'src/api/hooks/useApiTx';
+import {useApi} from 'context/ChainApiContext';
+import {useSnackbar} from 'context/SnackbarContext';
+import {InputLabel} from '@ui/library/InputLabel';
+import {decimalKeypad} from 'src/utils';
+import {useFormatBalance} from 'src/api/hooks/useFormatBalance';
+import {useBountiesSummary} from 'src/api/hooks/useBountiesSummary';
+import type {BN} from '@polkadot/util';
+import {formattedStringToBn} from 'src/api/utils/balance';
+
+export function AddBountyScreen({navigation}: {navigation: NavigationProp<AppStackParamList>}) {
+  const {colors} = useTheme();
+  const modalRef = useRef<Modalize>(null);
+  useEffect(() => {
+    modalRef.current?.open();
+  }, []);
+
+  const {data: bounty} = useBountiesSummary();
+  const {formatBalance, stringToBn} = useFormatBalance();
+
+  const [bountyTitle, setBountyTitle] = React.useState('');
+  const [bountyAllocation, SetBountyAllocation] = React.useState('');
+
+  const [account, setAccount] = React.useState<Account>();
+  const modelClose = () => modalRef.current?.close();
+  const startTx = useApiTx();
+  const {api} = useApi();
+  const snackbar = useSnackbar();
+
+  const {calculatedBountyBond, bountyAllocationBN, isBountyValid, isBountyTitleValid} = useMemo(() => {
+    if (bounty && bounty.bountyDepositBase) {
+      const calculatedBountyBond = calculateBountyBond(
+        bountyTitle,
+        formattedStringToBn(bounty.bountyDepositBase),
+        formattedStringToBn(bounty.dataDepositPerByte),
+      );
+      const bountyAllocationBN = stringToBn(bountyAllocation);
+      const isBountyValid = bountyAllocationBN
+        ? formattedStringToBn(bounty.bountyValueMinimum).gt(bountyAllocationBN)
+        : false;
+      const isBountyTitleValid = Number(bounty?.maximumReasonLength) <= Number(bountyTitle.length);
```

Comment:
Same here as well. Not checking what the variable name suggests.

Please change like following:
```suggestion
      const isBountyTitleValid = Number(bountyTitle.length) <= Number(bounty.maximumReasonLength);
```